src/mlpack/methods/adaboost/adaboost_fit_main.cpp

```diff
@@ -0,0 +1,180 @@
+using namespace std;
+using namespace arma;
+using namespace mlpack::adaboost;
+using namespace mlpack::tree;
+using namespace mlpack::perceptron;
+using namespace mlpack::util;
+
+// Program Name.
+BINDING_USER_NAME("AdaBoost Training");
+
+// Short description.
+BINDING_SHORT_DESC(
+    "Training AdaBoost model.");
+
+// Long description.
+BINDING_LONG_DESC("");
+
+// Example.
+BINDING_EXAMPLE(
+    IMPORT_EXT_LIB() + "\n" +
+    IMPORT_SPLIT() + "\n" +
+    IMPORT_THIS("adaboost") + "\n" +
+    GET_DATASET("X", "https://example.com") + "\n" +
+    GET_DATASET("y", "https://example.com") + "\n" +
+    SPLIT_TRAIN_TEST("X", "y", "X_train", "y_train", "X_test", "y_test", 
+    "0.2") + "\n" +
+    CREATE_OBJECT("model", "adaboost") + "\n" +
+    CALL_METHOD("model", "fit", "training", "X_train", "labels", "y_train"));
+
+// See also...
+BINDING_SEE_ALSO("AdaBoost on Wikipedia", "https://en.wikipedia.org/wiki/"
+        "AdaBoost");
+BINDING_SEE_ALSO("Improved boosting algorithms using confidence-rated "
+        "predictions (pdf)", "http://rob.schapire.net/papers/SchapireSi98.pdf");
+BINDING_SEE_ALSO("Perceptron", "#perceptron");
+BINDING_SEE_ALSO("Decision Stump", "#decision_stump");
+BINDING_SEE_ALSO("mlpack::adaboost::AdaBoost C++ class documentation",
+        "@doxygen/classmlpack_1_1adaboost_1_1AdaBoost.html");
+
+// Input for training.
+PARAM_MATRIX_IN("training", "Dataset for training AdaBoost.", "t");
+PARAM_UROW_IN("labels", "Labels for the training set.", "l");
+
+// Training options.
+PARAM_INT_IN("iterations", "The maximum number of boosting iterations to be run"
+    " (0 will run until convergence.)", "i", 1000);
+PARAM_DOUBLE_IN("tolerance", "The tolerance for change in values of the "
+    "weighted error during training.", "e", 1e-10);
+PARAM_STRING_IN("weak_learner", "The type of weak learner to use: "
+    "'decision_stump', or 'perceptron'.", "w", "decision_stump");
+
+// Loading/saving of a model.
+// PARAM_MODEL_IN(AdaBoostModel, "input_model", "Input AdaBoost model.", "m");
+PARAM_MODEL_OUT(AdaBoostModel, "output_model", "Output trained AdaBoost model.",
+    "M");
+
+void BINDING_FUNCTION(util::Params& params, util::Timers& timers)
+{
+  // Check input parameters and issue warnings/errors as necessary.
+  RequireOnlyOnePassed(params, {"training"});
+  RequireOnlyOnePassed(params, {"labels"});
+
+  // The weak learner must make sense.
+  RequireParamInSet<std::string>(params, "weak_learner",
+      { "decision_stump", "perceptron" }, true, "unknown weak learner type");
+
+  // Sanity check on iterations.
+  RequireParamValue<int>(params, "iterations", [](int x) { return x > 0; },
+      true, "invalid number of iterations specified");
+
+  // Sanity check on tolerance value.
+  RequireParamValue<double>(params, "tolerance", [](double x) { return x > 0; },
+      true, "invalid tolerance specified");
+
+  AdaBoostModel* m;
+
+  mat trainingData = std::move(params.Get<arma::mat>("training"));
+  m = new AdaBoostModel();
```

Comment:
```suggestion
  AdaBoostModel* m = new AdaBoostModel();
```
Minor simplification (goes with the comment above). :)